conf/script/pre_disruptive_upgrade.sh

```diff
@@ -0,0 +1,109 @@
+#!/bin/bash
+
+# Copyright (c) 2020 Seagate Technology LLC and/or its Affiliates
+#
+# This program is free software: you can redistribute it and/or modify it under the
+# terms of the GNU Affero General Public License as published by the Free Software
+# Foundation, either version 3 of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+# PARTICULAR PURPOSE. See the GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License along
+# with this program. If not, see <https://www.gnu.org/licenses/>. For any questions
+# about this software or licensing, please email opensource@seagate.com or
+# cortx-questions@seagate.com.
+
+set -eu -o pipefail
+
+BACKUP_SOURCE_DIR=/etc/cortx/ha
+BACKUP_DEST_DIR=/opt/seagate/cortx/ha_conf_backup
+consul=/usr/bin/consul
+HARE_DIR=/var/lib/hare
+
+backup_consul(){
+
+    echo "Taking the backup of consul"
+    consul_kv_dump=$HARE_DIR/consul-kv-dump.json
+    if [[ -f $consul_kv_dump ]]; then
+        cmd < $consul_kv_dump | xz > ${consul_kv_dump%.json}-$(date '+%Y%m%d%H%M%S').json.xz
+    fi
+    $consul kv export > $consul_kv_dump
+
+}
+
+backup_conf(){
+
+    echo "Taking the backup of ha conf directory"
+
+    [ -d $BACKUP_DEST_DIR ]  && {
+        rm -rf $BACKUP_DEST_DIR
+    }
+
+    mkdir -p $BACKUP_DEST_DIR
+
+    # Take the backup of configuration files
+    cp -r $BACKUP_SOURCE_DIR $BACKUP_DEST_DIR
+}
+
+cluster_standby_mode(){
+
+    echo "standby mode ON"
+
+    standby_mode=$(crm_standby -G  | awk '{print $3}' | cut -d '=' -f 2)
+
+    if [ "$standby_mode" == "on" ]
+    then
+        echo "cluster is already in a standby mode"
+    else
+        # Keep cluster (nodes and resources) on standby mode
+        /usr/sbin/pcs node standby --all --wait=10
+        [ $? == 0 ] || {
+
+            echo "some problem occured to keep cluster in standby mode, Hence exiting"
```

Comment:
It would be better to retry and give up after 2-3 tries?

Reply:
I agree, lets retry!
Also, will the update successful even if the standby fails?

Reply:
Sure. Done. So, wait for 30 secs and 3 retries. This way I have kept, let me know if any change needed.

Reply:
@Madhura-08 
Also, will the update successful even if the standby fails?

Reply:
No. It will wait for that particular seconds. And, after that it will return non-zero if standby fails. Otherwise zero.

Reply:
ok